de_templates/ingest/Ingest_AzureSql_SourceName/data_factory/adf_datasets.tf.jinja

```diff
@@ -0,0 +1,11 @@
+resource "azurerm_data_factory_custom_dataset" "ds_ex_AzureSql_{{ source_name }}Source" {
```

Comment:
I think we should take `AzureSql` as a template value, e.g. `source_type`. For now this the template is based around Azure SQL but we should expand this to support more types in future. (also applies to everywhere else we have `AzureSql`).
Also don't think it needs the 'Source' bit on the end.